crates/iota/src/key_identity.rs

```diff
@@ -41,19 +47,32 @@ impl Display for KeyIdentity {
         let v = match self {
             KeyIdentity::Address(x) => x.to_string(),
             KeyIdentity::Alias(x) => x.to_string(),
+            KeyIdentity::Name(x) => x.to_string(),
         };
         write!(f, "{}", v)
     }
 }
 
 /// Get the IotaAddress corresponding to this key identity.
 /// If no string is provided, then the current active address is returned.
-pub fn get_identity_address(
+pub async fn get_identity_address(
     input: Option<KeyIdentity>,
-    ctx: &mut WalletContext,
+    ctx: &WalletContext,
 ) -> Result<IotaAddress, Error> {
     if let Some(addr) = input {
-        get_identity_address_from_keystore(addr, ctx.config().keystore())
+        match addr {
+            KeyIdentity::Address(x) => Ok(x),
+            KeyIdentity::Alias(x) => Ok(*ctx.config().keystore().get_address_by_alias(x)?),
+            #[cfg(feature = "iota-names")]
+            KeyIdentity::Name(name) => {
+                let client = ctx.get_client().await?;
+                let entry = crate::name_commands::get_registry_entry(&name, &client).await?;
```

Comment:
Why not use the lookup endpoint?

Reply:
I also prefer that until we have a solution for local networks